lib/KettleRequest.js

```diff
@@ -433,15 +433,12 @@ kettle.request.http.successHandler = function (request, responseBody, responseOp
         request.res.set(field, headers[field]);
     }
 
-    if (request.req.method.toLowerCase() === "options") {
-        request.res.end();
-        return;
-    }
-
     if (typeof(responseBody) !== "string") {
         request.res.json(responseBody);
-    } else {
+    } else if (responseBody !== undefined) {
         request.res.send(responseBody);
+    } else {
+        request.res.end();
     }
 };
```

Comment:
I've made (or reproduced) a blunder here: the first if-condition is too open-ended, as it captures cases where `responseBody` is undefined. I think the obvious rewrite is to first check explicitly for the type `"object"`, then `"string"`, and keep the last branch as-is. This feels reasonable, as I don't think we'd ever want to process any other data types as a response body, so we can safely throw it away in the last branch whether or not it is undefined. Does this make sense?

Reply:
An oversight made clear by below commit: numbers are also valid JSON, so the first condition should test for `"object"` and `"number"`.

Reply:
Actually numbers are not valid JSON "texts" - strictly going by the (historical) spec, the top-level type of every valid JSON document should be "object" - https://stackoverflow.com/a/18419518 - even though this spec is in theory obsoleted, there is probably sufficient junk out there to be strict about it. We'll probably have to go with that since this method was faultily conceived from the start - a string is also a valid JSON document under the new spec which we will then proceed to faultily encode.
I suggest the more logical ordering
````
if (typeof(response) === "string") {
    request.send
} else if (response === undefined) {
   request.end
} else {
   request.json
}
````
It's a crock, but at least it's a backwards-compatible crock.